pallene/checker.lua

```diff
@@ -114,13 +114,19 @@ local function declare_type(type_name, cons)
     typedecl.declare(checker, "checker", type_name, cons)
 end
 
+declare_type("Mod_field", {
+    Function = {"loc", "decl"},
+    Var      = {"loc", "name", "typ"}
```

Comment:
A could certainly create a "fake `Decl`", i.e., a table containing `type` and `name`, if you think it would be better.

Reply:
Using a real `ast.Decl` might also be an option. Keep in mind that in the parser.lua, the Decl we create for the toplevel functions also is an "imaginary" one, that doesn't exist in the original program text.

Reply:
Sorry, I should've used clearer adjectives here, instead of "fake" and "real". I thought that using `ast.Decl` for something that doesn't correspond to a "declaration", in this case, an assignment, would be problematic (I actually thought about doing it). Since you think it's ok, I'm heavily considering doing so.

Reply:
You don't need to be sorry ;)

In the grammar, we use the Decl non-terminal for everything that is a name followed by an optional type annotation. That includes variable declarations and function arguments.

In the type checker we use references to these decl nodes to refer to the place where a given local variable is declared.

For toplevel declarations we also use Decls in some places but admittedly it's a bit more of a hack in this part.